packages/hint-manifest-scoped-urls/src/hint.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @fileoverview Checks if URLs in the manifest are in scope and accessible
+ */
+
+import { URL } from 'url';
+import { relative } from 'path';
+import { IHint, HintContext, NetworkData, ProblemLocation } from 'hint';
+import { ManifestEvents, ManifestParsed, Manifest } from '@hint/parser-manifest';
+import { debug as d } from '@hint/utils';
+
+import meta from './meta';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class ManifestScopedUrlsHint implements IHint {
+    public static readonly meta = meta;
+    public constructor(context: HintContext<ManifestEvents>) {
+        /**
+         * See if the `start_url` is accessible.
+         */
+        const startUrlAccessible = async (startUrl: string, resource: string, startUrllocation: ProblemLocation): Promise<boolean> => {
+            let networkData: NetworkData;
+
+            try {
+                networkData = await context.fetchContent(startUrl);
+            } catch (e) {
+                debug(`Failed to fetch ${startUrl}`);
+                const message = `Request failed for 'start_url'`;
+
+                context.report(resource, message, { location: startUrllocation });
+
+                return false;
+            }
+
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                const message = `Specified 'start_url' is not accessible. (status code: ${response.statusCode}).`;
+
+                context.report(resource, message, { location: startUrllocation });
+
+                return false;
+            }
+
+            return true;
+        };
+
+        /**
+         *
+         * @param property `property` to be found in the Manifest
+         * @param parsedContent Manifest
+         * @param resource
+         */
+        const manifestPropertyFound = (property: string, parsedContent: Manifest, resource: string): boolean => {
+            const found = parsedContent.hasOwnProperty(property);
+
+            if (!found) {
+                const message = `Property '${property}' not found in manifest file`;
+
+                context.report(resource, message);
+
+                return false;
+            }
+
+            return true;
+        };
+
+        /**
+         * Checks that the `start_url` is under the scope of
+         * the URL specified in the `scope`
+         * @param parsedContent
+         * @param resource
+         */
+        const startUrlInScope = (parsedContent: Manifest, resource: string, startUrllocation: ProblemLocation): boolean => {
+            const scope = parsedContent.scope || '/';
```

Comment:
At least according to https://developers.google.com/web/fundamentals/web-app-manifest/#scope
> If you don't include a scope in your manifest, then the default implied scope is the directory that your web app manifest is served from.

So we probably need to default `scope` to the directory `resource` is in if not specified (instead of `/`).

Reply:
Done.